### PR TITLE
Added unit tests.

### PR DESCRIPTION
**Pull request by aznashwan**
This PR adds some Pester unit tests to the powershell-yaml module functions.

### CODE REVIEW

File: Tests/powershell-yaml.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            return $true
+        }
+
+        if ($expected -is [Hashtable]) {
+            if ($got -isnot [Hashtable] -or ($expected.Count -ne $got.Count)) {
+                return $false
+            }
+
+            # iterate through all the keys:
+            $expected.Keys | % {
```
Comment (gabriel-samfira) [inline Tests/powershell-yaml.Tests.ps1]: does this work for nested hashtables?

File: Tests/powershell-yaml.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+$moduleName = "powershell-yaml"
+$modulePath = Resolve-Path "..\powershell-yaml.psm1"
+Import-Module $modulePath -DisableNameChecking
```
Comment (gabriel-samfira) [inline Tests/powershell-yaml.Tests.ps1]: Not sure ```-DisableNameChecking``` is needed. Can you check?

File: Tests/powershell-yaml.Tests.ps1 (hunk 1 above)
Comment (aznashwan) [inline Tests/powershell-yaml.Tests.ps1]: From my manual tests and reasoning about the thing; I'm confident it works well enough for our purposes.

File: Tests/powershell-yaml.Tests.ps1 (hunk 2 above)
Comment (aznashwan) [inline Tests/powershell-yaml.Tests.ps1]: Indeed it wasn't. Good catch.

File: Tests/powershell-yaml.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                    }
+                );
+                total = 4443.52;
+                comment = "I can't wait. To get that Cool Book."
+            }
+
+            $res = ConvertFrom-Yaml $testYaml
+
+            It "Should decode the YAML string as expected." {
+                Confirm-Equality $expected $res
```
Comment (gabriel-samfira) [inline Tests/powershell-yaml.Tests.ps1]: you forgot the assertion I think:

```powershell
Confirm-Equality $expected $res | ShouldBe $true
```

**Comment by gabriel-samfira** [comment]
Just one comment. Can you also add cases where the commandlets should fail?

For example:

https://github.com/cloudbase/powershell-yaml/blob/master/powershell-yaml.psm1#L212-L217

Otherwise looks good! thanks!

**Pull request merged**